### ISSUE TITLE
compile error with intell c++ compiler 2020

### ISSUE THREAD

**Issue by SchrodingerZhu** [randombit/botan#2260]
Error Hint:
```make
build/include/botan/internal/simd_avx2.h(156): error #3632: "target" attribute on special function is not supported
        void operator+=(const SIMD_8x32& other)
             ^

In file included from src/lib/block/shacal2/shacal2_avx2/shacal2_avx2.cpp(8):
build/include/botan/internal/simd_avx2.h(162): error #3632: "target" attribute on special function is not supported
        void operator-=(const SIMD_8x32& other)
             ^

In file included from src/lib/block/shacal2/shacal2_avx2/shacal2_avx2.cpp(8):
build/include/botan/internal/simd_avx2.h(168): error #3632: "target" attribute on special function is not supported
        void operator^=(const SIMD_8x32& other)
             ^

In file included from src/lib/block/shacal2/shacal2_avx2/shacal2_avx2.cpp(8):
build/include/botan/internal/simd_avx2.h(174): error #3632: "target" attribute on special function is not supported
        void operator|=(const SIMD_8x32& other)
             ^

In file included from src/lib/block/shacal2/shacal2_avx2/shacal2_avx2.cpp(8):
build/include/botan/internal/simd_avx2.h(180): error #3632: "target" attribute on special function is not supported
        void operator&=(const SIMD_8x32& other)
             ^

In file included from src/lib/block/shacal2/shacal2_avx2/shacal2_avx2.cpp(8):
build/include/botan/internal/simd_avx2.h(196): error #3632: "target" attribute on special function is not supported
        SIMD_8x32 operator~() const
```

**Comment by randombit** [comment]
Strange limitation. Try `--disable-modules=simd_avx2` and retry. If that works we just need to blacklist that code for ICC.

**Comment by SchrodingerZhu** [comment]
```
src/lib/rng/rdrand_rng/rdrand_rng.cpp(26): error #3550: target attribute not recognized
  BOTAN_FUNC_ISA("rdrnd")
```

**Comment by randombit** [comment]
Another weird one.

Amusingly the documentation for `target` attribute for Intel C++ just says "Specifies the target name. Possible values are:" ... without giving the values. ??? https://software.intel.com/en-us/cpp-compiler-developer-guide-and-reference-target

You might try editing the source to use "rdrand" instead of "rdrnd", but it's just a guess. If that works, let me know. Otherwise `--disable-modules=rdrand_rng`

**Comment by randombit** [comment]
RDRAND problem should be fixed in master now with 209e879. If you are seeing additional problems feel free to reopen this issue.
